go/runtime/txpool/txpool.go

```diff
@@ -460,20 +468,35 @@ func (t *txPool) checkTxBatch(ctx context.Context, rr host.RichRuntime) {
 		return
 	}
 
+	// Ensure block round is synced to storage.
+	waitSyncCtx, cancelWaitSyncCtx := context.WithTimeout(ctx, checkTxWaitRoundSyncedTimeout)
+	defer cancelWaitSyncCtx()
+
+	t.logger.Debug("ensuring block round is synced", "round", bi.RuntimeBlock.Header.Round)
+	if _, err = t.history.WaitRoundSynced(waitSyncCtx, bi.RuntimeBlock.Header.Round); err != nil {
+		// Block round isn't synced yet, so make sure the batch check is
+		// retried later to avoid aborting the runtime, as it is not its fault.
+		t.logger.Info("block round is not synced yet, retrying transaction batch check later",
+			"round", bi.RuntimeBlock.Header.Round,
+			"err", err,
+		)
+
+		go func() {
+			time.Sleep(checkTxWaitForNodeSyncRetryDelay)
```

Comment:
This can be removed if you wait for round synced for additional `checkTxWaitForNodeSyncRetryDelay`. Also, I don't see a point to wait in the background, as if the block is not synced, the next call will also fail. So we can remove go routine and just send to the channel to call this method again immediately. 

```go
	waitSyncCtx, cancelWaitSyncCtx := context.WithTimeout(ctx, checkTxWaitRoundSyncedTimeout)
	defer cancelWaitSyncCtx()

	t.logger.Debug("ensuring block round is synced", "round", bi.RuntimeBlock.Header.Round)
	if _, err = t.history.WaitRoundSynced(waitSyncCtx, bi.RuntimeBlock.Header.Round); err != nil {
		// Block round isn't synced yet, so make sure the batch check is
		// retried later to avoid aborting the runtime, as it is not its fault.
		t.logger.Info("block round is not synced yet, retrying transaction batch check later",
			"round", bi.RuntimeBlock.Header.Round,
			"err", err,
		)
		t.checkTxCh.In() <- struct{}{}
		return
	}
```